src/validation.cpp

```diff
@@ -1595,6 +1632,60 @@ PackageMempoolAcceptResult MemPoolAccept::AcceptPackage(const Package& package,
         }
     }
 
+    // Clear temporary coins that were pulled in during PreChecks or created in PackageAddTransaction.
+    CleanupTemporaryCoins();
+
+    // Now that we have determined which transactions to skip, do full validation for each transaction.
+    if (!quit_early) {
+        const bool rely_on_fees{linearized_package.LinearizeWithFees()};
+
+        for (const auto& tx : linearized_package.FilteredTxns()) {
+            const auto individual_fee_vsize = linearized_package.GetFeeAndVsize(tx);
+            TxValidationState placeholder_state;
+            if (rely_on_fees && Assume(individual_fee_vsize.has_value()) &&
+                !CheckFeeRate(individual_fee_vsize->second, individual_fee_vsize->first, placeholder_state)) {
+                // No need to validate if we know this transaction wouldn't meet feerate requirements.
+                // If we don't exit early here, we'll hit a low fee failure and quit early.
+                continue;
+            }
+            const auto subpackage = linearized_package.FilteredAncestorSet(tx);
+            if (!subpackage || subpackage->size() > 1) {
+                // No need to validate if we know this transaction would have missing inputs.
```

Comment:
Why does `            if (!subpackage || subpackage->size() > 1) {` imply missing inputs?

Reply:
I mean, you are clamping it to 1 on purpose: to keep existing behavior.

Reply:
Would it be better to squash the "add a PreChecks loop to linearize by fee and find obvious errors" and the "submit ancestor subpackages" commits maybe? There isn't a functional reason to do 1 without the other, I just thought maybe the diff would be smaller and easier to review. But maybe it's just confusing?

Reply:
> I mean, you are clamping it to 1 on purpose: to keep existing behavior.

o yeah true

Reply:
I've squashed the 2 commits